### PR TITLE
Device log migrations

### PR DESCRIPTION
**Pull request by benrudolph**
@czue migrations for couchdbs I think will look something like this:
1. Define a new mgmt command that subclasses the `MigrateCommand` and specify the `MigrationPillow` to use
2. Deploy the pillow and have it run. It should be a noop until `is_migrating` is set to `False`
3. Run the `couch_migrate_XXX` command to seed the new database
4. Set the `switched_dbs` flag to `True` when ready to fully switch over. This will cause the pillow to stop processing deletions.
5. Delete old database and :pray: 

https://github.com/dimagi/pillowtop/pull/125

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/pillows/devicelogs.py, line 5 - E231 missing whitespace after ':'
* corehq/pillows/devicelogs.py, line 5 - E901 SyntaxError: invalid syntax
* corehq/pillows/devicelogs.py, line 12 - W391 blank line at end of file
* corehq/pillows/migration.py, line 3 - F401 'get_deleted_doc_types' imported but unused
* corehq/pillows/migration.py, line 25 - F821 undefined name 'changed_dict'
* corehq/pillows/migration.py, line 26 - F821 undefined name 'changed_dict'
* corehq/pillows/migration.py, line 29 - F821 undefined name 'changes_dic'

File: corehq/pillows/migration.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from pillowtop.listener import BasicPillow
+from couchforms.models import XFormInstanceDevicelog, XFormInstance
+from .utils import get_deleted_doc_types
```
Comment (dimagimon) [inline corehq/pillows/migration.py]: F401 'get_deleted_doc_types' imported but unused

File: corehq/pillows/migration.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    """
+
+    # Override this to determine what constitutes a deleted document
+    # is_deleted = lambda doc: False
+
+    # Override this function to get the db instance you'd like to migrate to
+    dest_db = lambda: None
+
+    def change_trigger(self, changes_dict):
+        if changed_dict.get('deleted', False):
```
Comment (dimagimon) [inline corehq/pillows/migration.py]: F821 undefined name 'changed_dict'

File: corehq/pillows/migration.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            self.dest_db().delete_doc(changed_dict['id'])
```
Comment (dimagimon) [inline corehq/pillows/migration.py]: F821 undefined name 'changed_dict'

File: corehq/pillows/migration.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            return None
+        # This seems to defe the purpose of making it a mixin...
+        super(MigrationPillow, self).change_trigger(changes_dic)
```
Comment (dimagimon) [inline corehq/pillows/migration.py]: F821 undefined name 'changes_dic'

**Comment by benrudolph** [comment]
Random thoughts: It would be nice if the migrate copy code and the migrate pillow code should use the same config so there's less chance of bugs where the pillow doesn't pick up the same docs as the copy. 

Also would be cool if the seq is was saved to some kind table and then that would trigger the pillow to start from that seq Id. It seems hacky to copy the docs, save the seq id to a file, start up a pillow with that seq id via command line. Would be nice if it were a bit more automated/fluid but I'm not entirely sure what that looks like

**Comment by dannyroberts** [comment]
@benrudolph sorry for the much delayed response. Do you want to go over this with me sometime?

**Comment by benrudolph** [comment]
Yea that'd be great. Still something I want to do, but the process is quite
clunky. Would be nice to brainstorm some better ways of accomplishing this

**Comment by czue** [comment]
@benrudolph @dannyroberts is there any plan for this or are we just dropping until @benrudolph is back from vacation

**Comment by dannyroberts** [comment]
I have a reminder to go over it with ben when he gets back from India; I
think we're dropping until then

File: corehq/apps/cleanup/management/commands/couch_migrate.py [hunk 5]
```diff
@@ -1,17 +1,22 @@
 import json
-from django.core.management.base import LabelCommand, CommandError
+from datetime import datetime
```
Comment (dimagimon) [inline corehq/apps/cleanup/management/commands/couch_migrate.py]: F401 'datetime' imported but unused

**Comment by dannyroberts** [comment]
(not mergeable)

**Comment by benrudolph** [comment]
that's ok i don't think this is quite read for the spot light. gotta iterate on it a bit more

**Comment by czue** [comment]
is this superceded by the user db workflow?

**Comment by benrudolph** [comment]
yup

**Comment by benrudolph** [comment]
we should be able to migrate fixtures devicelogs rather painlessly now

**Comment by czue** [comment]
let's get that next on the backlog. @dannyroberts

**Comment by dannyroberts** [comment]
:+1: very excite!

**Pull request closed without merging**